tools/cloud-build/daily-tests/ansible_playbooks/tasks/check_stockout.yml

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+
+- name: Assert variables are defined
+  ansible.builtin.assert:
+    that:
+    - project is defined
+
+- name: Check compute stockout
+  changed_when: false
+  register: stockout
+  until: stockout.rc == 0
+  retries: 5
+  delay: 10
+  vars:
+    stockout_string: Region does not currently have sufficient capacity for the requested resources.
+  ansible.builtin.command: >-
+    gcloud logging --project {{ project }} read
+    'protoPayload.response.error.errors.message="{{ stockout_string }}" AND protoPayload.request.perInstanceProperties.key=~".*{{ slurm_cluster_name }}.*"'
+    --flatten="protoPayload.request.perInstanceProperties"
+    --format='table(protoPayload.request.perInstanceProperties.key:label=INSTANCE_ID,protoPayload.response.error.errors.message.list():label=ERROR_MESSAGE)'
+    --limit=5
+  when: slurm_cluster_name is defined
+
+- name: Log stockout error
+  ansible.builtin.debug:
+    msg: |
+      "Abbreviated listing of nodes that could not be created:"
+      "{{ stockout.stdout }}"
```

Comment:
Can you provide an example snippet of what this looks like in practice in the PR description. 

Also include information in description about testing performed.

Reply:
It should look something like this:

```
INFO 2023-10-18T20:57:29.558118841Z Step #2 - "hcls": "INSTANCE_ID ERROR_MESSAGE
INFO 2023-10-18T20:57:29.558120127Z Step #2 - "hcls": hcls29c035-gpu-ghpc-5 ['Region does not currently have sufficient capacity for the requested resources.']
INFO 2023-10-18T20:57:29.558121179Z Step #2 - "hcls": hcls29c035-gpu-ghpc-2 ['Region does not currently have sufficient capacity for the requested resources.']
INFO 2023-10-18T20:57:29.558122035Z Step #2 - "hcls": hcls29c035-gpu-ghpc-10 ['Region does not currently have sufficient capacity for the requested resources.']
INFO 2023-10-18T20:57:29.558122924Z Step #2 - "hcls": hcls29c035-gpu-ghpc-9 ['Region does not currently have sufficient capacity for the requested resources.']
INFO 2023-10-18T20:57:29.558123651Z Step #2 - "hcls": hcls29c035-gpu-ghpc-14 ['Region does not currently have sufficient capacity for the requested resources.']"
```

The error message won't be in square brackets though.

Reply:
This has been updated, the output looks slightly different:

```
Step #2 - "hcls": TASK [Check for stockout errors] ***********************************************
Step #2 - "hcls": included: /workspace/tools/cloud-build/daily-tests/ansible_playbooks/tasks/check_stockout.yml for 130.211.102.173
Step #2 - "hcls": 
Step #2 - "hcls": TASK [Assert variables are defined] ********************************************
Step #2 - "hcls": ok: [130.211.102.173 -> localhost] => {
Step #2 - "hcls":     "changed": false
Step #2 - "hcls": }
Step #2 - "hcls": 
Step #2 - "hcls": MSG:
Step #2 - "hcls": 
Step #2 - "hcls": All assertions passed
Step #2 - "hcls": 
Step #2 - "hcls": TASK [Check logs for stockout on compute nodes] ********************************
Step #2 - "hcls": ok: [130.211.102.173 -> localhost] => (item=Region does not currently have sufficient capacity for the requested resources.)
Step #2 - "hcls": ok: [130.211.102.173 -> localhost] => (item=No eligible zone could be found in this region for given properties)
Step #2 - "hcls": 
Step #2 - "hcls": TASK [Log compute stockout error] **********************************************
Step #2 - "hcls": skipping: [130.211.102.173] => (item=Region does not currently have sufficient capacity for the requested resources.) 
Step #2 - "hcls": ok: [130.211.102.173 -> localhost] => (item=No eligible zone could be found in this region for given properties) => {}
Step #2 - "hcls": 
Step #2 - "hcls": MSG:
Step #2 - "hcls": 
Step #2 - "hcls": "Abbreviated listing of nodes that could not be created:"
Step #2 - "hcls": "INSTANCE_ID                   ERROR_MESSAGE
Step #2 - "hcls": hcls7636d5-gpu-ghpc-9  No eligible zone could be found in this region for given properties
Step #2 - "hcls": hcls7636d5-gpu-ghpc-1  No eligible zone could be found in this region for given properties
Step #2 - "hcls": hcls7636d5-gpu-ghpc-3  No eligible zone could be found in this region for given properties
Step #2 - "hcls": hcls7636d5-gpu-ghpc-5  No eligible zone could be found in this region for given properties
Step #2 - "hcls": hcls7636d5-gpu-ghpc-8  No eligible zone could be found in this region for given properties"
```